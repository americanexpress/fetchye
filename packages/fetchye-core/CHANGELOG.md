# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.1.0](https://github.com/americanexpress/fetchye/compare/v1.0.0...v1.1.0) (2021-10-07)


### Bug Fixes

* **defaultFetcher:** Avoid known hang in node-fetch ([d33399c](https://github.com/americanexpress/fetchye/commit/d33399c71c6809a868c8f6b928ed349d995d8f6a))
* **jsonError:** update to handle NO CONTENT response ([030f886](https://github.com/americanexpress/fetchye/commit/030f8863de37fef253fcac477d8f30c55bd88cf7))
* **makeServerFetchye:** mask errors from server side fetching ([#49](https://github.com/americanexpress/fetchye/issues/49)) ([3503836](https://github.com/americanexpress/fetchye/commit/3503836ea914ff4877af1682c7c672e834286264))





## [1.0.4](https://github.com/americanexpress/fetchye/compare/v1.0.0...v1.0.4) (2021-08-10)


### Bug Fixes

* **defaultFetcher:** Avoid known hang in node-fetch ([d33399c](https://github.com/americanexpress/fetchye/commit/d33399c71c6809a868c8f6b928ed349d995d8f6a))
* **jsonError:** update to handle NO CONTENT response ([030f886](https://github.com/americanexpress/fetchye/commit/030f8863de37fef253fcac477d8f30c55bd88cf7))
* **makeServerFetchye:** mask errors from server side fetching ([#49](https://github.com/americanexpress/fetchye/issues/49)) ([3503836](https://github.com/americanexpress/fetchye/commit/3503836ea914ff4877af1682c7c672e834286264))





## [1.0.3](https://github.com/americanexpress/fetchye/compare/v1.0.0...v1.0.3) (2021-07-14)


### Bug Fixes

* **defaultFetcher:** Avoid known hang in node-fetch ([d33399c](https://github.com/americanexpress/fetchye/commit/d33399c71c6809a868c8f6b928ed349d995d8f6a))
* **jsonError:** update to handle NO CONTENT response ([030f886](https://github.com/americanexpress/fetchye/commit/030f8863de37fef253fcac477d8f30c55bd88cf7))





## [1.0.2](https://github.com/americanexpress/fetchye/compare/v1.0.0...v1.0.2) (2021-03-08)


### Bug Fixes

* **jsonError:** update to handle NO CONTENT response ([030f886](https://github.com/americanexpress/fetchye/commit/030f8863de37fef253fcac477d8f30c55bd88cf7))





## 1.0.1 (2020-11-17)


### Features

* **fetchye:** prepare for initial release [skip ci] ([#28](https://github.com/americanexpress/fetchye/issues/28)) ([2b212df](https://github.com/americanexpress/fetchye/commit/2b212df8fab4405e2b7c51ad687a280cfe27ebbd))





# 1.0.0 (2020-11-12)


### Features

* **fetchye:** prepare for initial release [skip ci] ([#28](https://github.com/americanexpress/fetchye/issues/28)) ([2b212df](https://github.com/americanexpress/fetchye/commit/2b212df8fab4405e2b7c51ad687a280cfe27ebbd))





# [1.0.0-beta.6](https://github.com/americanexpress/fetchye/compare/v1.0.0-beta.5...v1.0.0-beta.6) (2020-11-12)

**Note:** Version bump only for package fetchye-core





# [1.0.0-beta.5](https://github.com/americanexpress/fetchye/compare/v1.0.0-beta.3...v1.0.0-beta.5) (2020-11-04)



# 1.0.0-beta.4 (2020-11-03)

**Note:** Version bump only for package fetchye-core





# [1.0.0-beta.4](https://github.com/americanexpress/fetchye/compare/v1.0.0-beta.3...v1.0.0-beta.4) (2020-11-03)

**Note:** Version bump only for package fetchye-core
