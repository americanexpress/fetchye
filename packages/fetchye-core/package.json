{
  "name": "fetchye-core",
  "version": "1.0.0-beta.4",
  "description": "Core components of the Fetchye library",
  "license": "Apache-2.0",
  "main": "lib/index.js",
  "module": "es/index.js",
  "files": [
    "lib",
    "es"
  ],
  "scripts": {
    "clean": "rimraf lib && rimraf es",
    "prebuild": "yarn clean",
    "build:cjs": "cross-env BABEL_ENV=commonjs babel --extensions .js,.jsx ./src --out-dir lib",
    "build:esm": "babel --extensions .js,.jsx ./src --out-dir es",
    "build": "yarn build:cjs && yarn build:esm"
  },
  "keywords": [
    "fetchye-core",
    "fetchye"
  ],
  "contributors": [
    "James Singleton <James.Singleton1@aexp.com> https://github.com/JamesSingleton)",
    "Jimmy King  <Jimmy.King@aexp.com> (https://github.com/10xLaCroixDrinker)",
    "Jonathan Adshead <Jonathan.Adshead@aexp.com> (https://github.com/JAdshead)",
    "Michael Tobia <Michael.M.Tobia@aexp.com> (https://github.com/Francois-Esquire)",
    "Nelly Kiboi <Nelly.J.Kiboi@aexp.com> (https://github.com/nellyk)",
    "Ruben Casas <ruben.casas@aexp.com> (https://github.com/infoxicator)"
  ],
  "dependencies": {
    "create-shared-react-context": "^1.0.3"
  },
  "devDependencies": {
    "@babel/cli": "^7.12.1",
    "@babel/core": "^7.12.3",
    "@testing-library/react-hooks": "^3.4.2",
    "babel-preset-amex": "^3.4.1",
    "cross-env": "^7.0.2",
    "react-test-renderer": "^16.14.0",
    "rimraf": "^3.0.2"
  },
  "peerDependencies": {
    "react": "^16.14.0",
    "react-dom": "^16.14.0"
  }
}
