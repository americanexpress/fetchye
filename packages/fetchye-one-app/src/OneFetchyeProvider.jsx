/*
 * Copyright 2020 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { FetchyeReduxProvider } from 'fetchye-redux-provider';
import invariant from 'invariant';
import OneCache from './OneCache';

const OneFetchyeProvider = ({
  children,
  fetcher,
  fetchClient,
  equalityChecker,
  cache,
}) => {
  invariant(
    !cache,
    '"cache" is not a permitted argument for OneFetchyeProvider, please use the FetchyeReduxProvider provided by fetchye-redux-provider instead: https://github.com/americanexpress/fetchye#fetchyereduxprovider'
  );
  invariant(
    !equalityChecker,
    '"equalityChecker" is not a permitted argument for OneFetchyeProvider, please use the FetchyeReduxProvider provided by fetchye-redux-provider instead: https://github.com/americanexpress/fetchye#fetchyereduxprovider'
  );
  invariant(
    !fetcher,
    '"fetcher" is not a permitted argument for OneFetchyeProvider, please use the FetchyeReduxProvider provided by fetchye-redux-provider instead: https://github.com/americanexpress/fetchye#fetchyereduxprovider'
  );
  return (
    <FetchyeReduxProvider
      cache={OneCache()}
      fetchClient={fetchClient}
    >
      {children}
    </FetchyeReduxProvider>
  );
};

OneFetchyeProvider.propTypes = {
  cache: PropTypes.shape({
    reducer: PropTypes.func.isRequired,
    getCacheByKey: PropTypes.func.isRequired,
    cacheSelector: PropTypes.func.isRequired,
  }).isRequired,
  equalityChecker: PropTypes.func,
  fetchClient: PropTypes.func,
  fetcher: PropTypes.func,
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
};

export default OneFetchyeProvider;
