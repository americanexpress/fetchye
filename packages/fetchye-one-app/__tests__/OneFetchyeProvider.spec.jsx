/*
 * Copyright 2020 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React, { useContext } from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { FetchyeContext } from 'fetchye-core';
import OneFetchyeProvider from '../src/OneFetchyeProvider';

const store = createStore((state) => state, { initialState: {} });

global.fetch = () => {};

describe('OneFetchyeProvider', () => {
  it('creates fetchye context', () => {
    let contextResult;
    render(
      <Provider store={store}>
        <OneFetchyeProvider>
          {React.createElement(() => {
            contextResult = useContext(FetchyeContext);
            return <fake-element />;
          })}
        </OneFetchyeProvider>
      </Provider>
    );
    expect(contextResult).toMatchInlineSnapshot(`
      Object {
        "cache": Object {
          "cacheSelector": [Function],
          "getCacheByKey": [Function],
          "reducer": [Function],
        },
        "defaultFetcher": [Function],
        "dispatch": [Function],
        "fetchClient": [Function],
        "useFetchyeSelector": [Function],
      }
    `);
  });

  it('throws when cache provided', () => {
    expect(() => render(
      <OneFetchyeProvider cache="boom">
        <fake-element />
      </OneFetchyeProvider>
    )).toThrowErrorMatchingSnapshot();
  });

  it('throws when equalityChecker provided', () => {
    expect(() => render(
      <OneFetchyeProvider equalityChecker="boom">
        <fake-element />
      </OneFetchyeProvider>
    )).toThrowErrorMatchingSnapshot();
  });

  it('throws when fetcher provided', () => {
    expect(() => render(
      <OneFetchyeProvider fetcher="boom">
        <fake-element />
      </OneFetchyeProvider>
    )).toThrowErrorMatchingSnapshot();
  });
});
