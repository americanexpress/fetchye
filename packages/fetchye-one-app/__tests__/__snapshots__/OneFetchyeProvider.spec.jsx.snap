// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OneFetchyeProvider throws when cache provided 1`] = `"\\"cache\\" is not a permitted argument for OneFetchyeProvider, please use the FetchyeReduxProvider provided by fetchye-redux-provider instead: https://github.com/americanexpress/fetchye#fetchyereduxprovider"`;

exports[`OneFetchyeProvider throws when equalityChecker provided 1`] = `"\\"equalityChecker\\" is not a permitted argument for OneFetchyeProvider, please use the FetchyeReduxProvider provided by fetchye-redux-provider instead: https://github.com/americanexpress/fetchye#fetchyereduxprovider"`;

exports[`OneFetchyeProvider throws when fetcher provided 1`] = `"\\"fetcher\\" is not a permitted argument for OneFetchyeProvider, please use the FetchyeReduxProvider provided by fetchye-redux-provider instead: https://github.com/americanexpress/fetchye#fetchyereduxprovider"`;
